src/app/fdctl/configure/workspace.c

```diff
@@ -139,6 +146,20 @@ static void buf( void * pod, char * fmt, void * value, ulong sz, ... ) {
     FD_LOG_ERR(( "failed to initialize workspace" ));
 }
 
+
+
+static inline uchar *
+read_key( char const * key_path,
+          uchar      * key       ) {
+  FILE * key_file = fopen( key_path, "r" );
+  if( FD_UNLIKELY( !key_file ) ) FD_LOG_ERR(( "Opening key file (%s) failed", key_path ));
+
+  if( FD_UNLIKELY( 1!=fscanf( key_file, "[%hhu", &key[0] ) ) ) FD_LOG_ERR(( "parsing key file failed at pos=0" ));
+  for( ulong i=1UL; i<64UL; i++ ) if( FD_UNLIKELY( 1!=fscanf( key_file, ",%hhu", &key[i] ) ) ) FD_LOG_ERR(( "parsing key file failed at pos=%lu", i ));
```

Comment:
Can you scanf the last "]" and also then check for feof to make sure the file is valid.